packages/core/test/model.test.ts

```diff
@@ -199,6 +208,233 @@ describe('createModel', () => {
     expect(updatedState.context.age).toEqual(42);
   });
 
+  it('can model actions', () => {
+    const userModel = createModel(
+      {
+        name: 'David',
+        age: 30
+      },
+      {
+        actions: {
+          greet: (message: string) => ({ message })
+        }
+      }
+    );
+
+    userModel.createMachine({
```

Comment:
So I guess that the `createMachine<typeof userModel>` should work as well but fixing that can wait for @devanshj's feedback because I'm unsure how this should be refactored to allow that in the strict mode.

Reply:
Right. `model.createMachine` does not give me type errors.